sstables/mx/reader.cc

```diff
@@ -1737,6 +1737,13 @@ mutation_reader make_reader(
             value_or_reference(std::move(slice)), std::move(trace_state), fwd, fwd_mr, monitor);
 }
 
+/// a reader which does not support seeking to given position.
+///
+/// unlike max_sstable_mutation_reader which allows positioned read,
+/// mx_crawling_sstable_mutation_reader only supports sequential read. It is designed
+/// to be used in conditions where:
+/// - the index is not reliable, or
+/// - the consumer reads the whole sstable
 class mx_crawling_sstable_mutation_reader : public mp_row_consumer_reader_mx {
```

Comment:
I think we should also rename this to `mx_sstable_full_scan_reader`. The crawling reader name is too obscure it seems.

Reply:
i will do this in a dedicated commit in this changeset.